src/frontend/game/components/ChangeName.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'preact/hooks';
+import { type FunctionalComponent } from 'preact';
+import { LineInput } from 'src/frontend/components/LineInput.tsx';
+import { useClientContext } from 'src/frontend/game/ClientContextProvider.tsx';
+import { validName } from 'src/agnostic/validation.ts';
+
+export const ChangeName: FunctionalComponent = () => {
+	let [clientContext, dispatchGameEvent] = useClientContext();
```

Comment:
`clientContext` represents the frontend's global state. it has 2 fields, `gameState`, which is syned with the server, and `clientState`, which is specific to this client and only stored on the frontend. to get access to the `clientContext` in this project we use the `useClientContext` hook, which returns an array of 3 elements: the current client context, a function to dispatch game events to the server, and a function to set the client state (which is not shown in this code because it's not needed for this component).